### PR TITLE
[dev][No PBI] - Fixed clear all disabling issue on filter rails when applying filter buttons

### PR DESCRIPTION
**Pull request by fantaJinMode**

### CODE REVIEW

File: src/surfaces/filtersRail/filtersRail.jsx [hunk 1]
```diff
@@ -290,7 +290,7 @@ function FiltersRail(props) {
                             <div className={actionButtonsClasses}>
                                 <A
                                     className={clearFiltersClasses}
-                                    disable={!filterOptions.isDirty || !filterOptions.isFiltering}
+                                    disable={!filterOptions.isDirty && !filterOptions.isFiltering}
```
Comment (groberts314) [inline src/surfaces/filtersRail/filtersRail.jsx:293]: I knew this was a bit tricky and hard to follow! =)
What if we got rid of the NOTs and made it an OR -- would that be easier to read/understand?
```JavaScript
disable={filterOptions.isDirty || filterOptions.isFiltering}
```

**Review by groberts314 — APPROVED** [review]
One small non-blocking suggestion to improve readability.

**Comment by groberts314** [comment]
@fantaJinMode Have you tested this?
Is it worth making it functional on the Demo Page / Demo Drawer?

File: src/surfaces/filtersRail/filtersRail.jsx (hunk 1 above)
Comment (fantaJinMode) [reply src/surfaces/filtersRail/filtersRail.jsx:293]: Yeah, you previously commented on one of the PR about that same condition. Stupid of me, should have double checked it 😞 It's confusing as well =)

**Comment by fantaJinMode** [comment]
> @fantaJinMode Have you tested this? Is it worth making it functional on the Demo Page / Demo Drawer?

Yeah, I tested it on the demo drawer. It's functional on demo drawer but not on demo page. May be, I still have tomorrow last day I can work on it? Any thoughts? @groberts314

**Comment by groberts314** [comment]
> > @fantaJinMode Have you tested this? Is it worth making it functional on the Demo Page / Demo Drawer?
> 
> Yeah, I tested it on the demo drawer. It's functional on demo drawer but not on demo page. May be, I still have tomorrow last day I can work on it? Any thoughts? @groberts314

Yeah, if you'd like to do some additional work on either the documentation demos (and/or maybe some nice unit tests), that would be great.  I can hold off merging.  Alternatively, if you're pretty sure this is the one, I can merge this and release 10.9.3 with this fix, and then you can do additional stuff for the docs and/or unit tests in another PR.  I'm good either way.

**Comment by fantaJinMode** [comment]
> > > @fantaJinMode Have you tested this? Is it worth making it functional on the Demo Page / Demo Drawer?
> > 
> > 
> > Yeah, I tested it on the demo drawer. It's functional on demo drawer but not on demo page. May be, I still have tomorrow last day I can work on it? Any thoughts? @groberts314
> 
> Yeah, if you'd like to do some additional work on either the documentation demos (and/or maybe some nice unit tests), that would be great. I can hold off merging. Alternatively, if you're pretty sure this is the one, I can merge this and release 10.9.3 with this fix, and then you can do additional stuff for the docs and/or unit tests in another PR. I'm good either way.

As discussed on our scrum, let's merge this PR and will do the additional work related to demo page on another PR
